main/integration-tests/__test__/api-tests/appstream-egress-enabled/workspace-service-catalog/connections/create-url-connection.test.js

```diff
@@ -0,0 +1,103 @@
+/*
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License").
+ *  You may not use this file except in compliance with the License.
+ *  A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ *  or in the "license" file accompanying this file. This file is distributed
+ *  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+const fetch = require('node-fetch');
+const { runSetup } = require('../../../../../support/setup');
+
+describe('Create URL scenarios', () => {
+  let setup;
+  let adminSession;
+
+  beforeAll(async () => {
+    setup = await runSetup();
+    adminSession = await setup.defaultAdminSession();
+  });
+
+  afterAll(async () => {
+    await setup.cleanup();
+  });
+
+  // These tests assume workspaces have already been created in the SWB environment
+  // TODO: Create a new workspace during these tests, and terminate once done (dependent on GALI-1093)
+  describe('Create AppStream URL', () => {
+    it('should return AppStream URL SageMaker', async () => {
+      const envId = '36f22de5-fefb-4f62-8ab4-e99ac4387ad4';
+      const connectionId = 'id-0';
+      const applicationName = 'Firefox';
+      const sagemakerUrlPrefix = 'https://basicnotebookinstance';
+      const preAuthStreamingUrl = 'https://appstream2.us-east-1.aws.amazon.com/authenticate?parameters=';
+      const redirectStreamingUrl = `appstream2.us-east-1.aws.amazon.com/#/streaming/?reference=fleet%2Finitial-stack-1629237287942-ServiceWorkbenchFleet&app=${applicationName}`;
+
+      const retVal = await adminSession.resources.workspaceServiceCatalogs
+        .workspaceServiceCatalog(envId)
+        .connections()
+        .connection(connectionId)
+        .createUrl();
+
+      const response = await fetch(retVal.url);
+      const redirectResponse = response[Object.getOwnPropertySymbols(response)[1]];
```

Comment:
I would be interested in hearing your thoughts on this. Can we make the test simpler to read by moving the checks closer to the action? Example.

```
// BUILD
const envId = '36f22de5-fefb-4f62-8ab4-e99ac4387ad4';
const connectionId = 'id-0';
const applicationName = 'Firefox';
const sagemakerUrlPrefix = 'https://basicnotebookinstance';
const preAuthStreamingUrl = 'https://appstream2.us-east-1.aws.amazon.com/authenticate?parameters=';
const redirectStreamingUrl = `appstream2.us-east-1.aws.amazon.com/#/streaming/?reference=fleet%2Finitial-stack-1629237287942-ServiceWorkbenchFleet&app=${applicationName}`;

// OPERATE
const retVal = await adminSession.resources.workspaceServiceCatalogs
    .workspaceServiceCatalog(envId)
    .connections()
    .connection(connectionId)
    .createUrl();

// CHECK
expect(retVal.appstreamDestinationUrl).toContain(sagemakerUrlPrefix);
expect(retVal.id).toEqual(connectionId);
expect(retVal.url).toContain(preAuthStreamingUrl);

// OPERATE
const response = await fetch(retVal.url);
const redirectResponse = response[Object.getOwnPropertySymbols(response)[1]];

// CHECK
expect(redirectResponse.url).toContain(redirectStreamingUrl);
  ```

Reply:
Yup, good idea. I'll make the change.